__tests__/rateRoutes.test.js

```diff
@@ -0,0 +1,94 @@
+import sinon from 'sinon';
+import chai from 'chai';
+import chaiHttp from 'chai-http';
+import models from '../database/models';
+import app from '../index';
+import mockRates from './mockData/mockRates';
+
+chai.use(chaiHttp);
+
+const { Rating, Article } = models;
+
+const url = '/api/v1/articles/ratings';
+const loginUrl = '/api/v1/auth/signin';
+const { expect } = chai;
+let validToken;
+
+describe('Rating test', () => {
+  before((done) => {
+    const userDetails = {
+      email: 'johndoe@test.com',
+      password: 'Password123'
+    };
+    chai
+      .request(app)
+      .post(loginUrl)
+      .send(userDetails)
+      .end((err, res) => {
+        const { token } = res.body.user;
+        validToken = token;
+        done();
+      });
+  });
+  describe('Create Rating', () => {
+    it('should throw a 500 when an error occurs on the server', (done) => {
+      const stub = sinon.stub(Rating, 'create').rejects(new Error('Foreign Key constraint'));
+      chai
+        .request(app)
+        .post(url)
+        .set('Authorization', `Bearer ${validToken}`)
+        .send({
+          id: 1,
+          rate: 3
+        })
+        .end((err, res) => {
+          expect(res.status).to.eql(500);
+          stub.restore();
+          done();
+        });
+    });
+
+    it('should create a rating successfully', (done) => {
+      chai
+        .request(app)
+        .post(`${url}`)
+        .set('Authorization', `Bearer ${validToken}`)
+        .send(mockRates[0])
+        .end((err, res) => {
+          expect(res.status).to.equal(201);
+          expect(res.body.data.message).to.be.eql('Your rating has been added successfully');
+          expect(res.body.data).to.have.property('articleId');
+          expect(res.body.data).to.have.property('rate');
+          expect(res.body.data).to.have.property('firstName');
+          expect(res.body.data).to.have.property('lastName');
+          done();
+        });
+    });
+    it('should throw error when author tries to rate his article', (done) => {
+      chai
+        .request(app)
+        .post(`${url}`)
+        .set('Authorization', `Bearer ${validToken}`)
+        .send(mockRates[1])
+        .end((err, res) => {
+          expect(res.status).to.equal(400);
```

Comment:
Great implementation! I like that you've considered the edge case where an author tries to rate his own article